crates/wash-cli/tests/wash_wit.rs

```diff
@@ -91,3 +92,76 @@ async fn test_wash_wit_wkg_override() {
         .expect("Failed to fetch dependencies");
     assert!(fetch_status.success(), "Failed to fetch dependencies");
 }
+
+/// Succeed on various "extended" configuration pull overrides (valid sources)
+#[tokio::test]
+#[cfg_attr(not(can_reach_ghcr_io), ignore = "ghcr.io is not reachable")]
+#[cfg_attr(not(can_reach_github_com), ignore = "github.com is not reachable")]
+async fn test_wash_wit_extended_valid_sources() -> Result<()> {
+    let test_setup = load_fixture("integrated-wkg-extended").await.unwrap();
+
+    // Run `wit fetch` (an alias for wit deps)
+    let fetch_status = test_setup
+        .base_command()
+        .args(["wit", "fetch"])
+        .stdout(std::process::Stdio::inherit())
+        .stderr(std::process::Stdio::inherit())
+        .status()
+        .await
+        .expect("Failed to fetch dependencies");
+    assert!(fetch_status.success(), "Failed to fetch dependencies");
+
+    // Ensure the expected directories are present
+    let wit_output_dir = test_setup.project_dir.join("wit");
+    assert!(tokio::fs::metadata(&wit_output_dir)
+        .await
+        .is_ok_and(|m| m.is_dir()));
+    let deps_dir = wit_output_dir.join("deps");
+    let wasmcloud_bus_dir = deps_dir.join("wasmcloud-bus-1.0.0");
+    let test_components_dir = deps_dir.join("test-components-testing-0.1.0");
+    assert!(tokio::fs::metadata(&wasmcloud_bus_dir)
+        .await
+        .is_ok_and(|m| m.is_dir()));
+    assert!(tokio::fs::metadata(&test_components_dir)
+        .await
+        .is_ok_and(|m| m.is_dir()));
+    assert!(tokio::fs::metadata(&wasmcloud_bus_dir)
+        .await
+        .is_ok_and(|m| m.is_dir()));
+    assert!(tokio::fs::metadata(&wasmcloud_bus_dir)
+        .await
+        .is_ok_and(|m| m.is_dir()));
+    let wasmcloud_bus_wit = wasmcloud_bus_dir.join("package.wit");
+    let test_components_wit = test_components_dir.join("package.wit");
+    assert!(tokio::fs::metadata(&wasmcloud_bus_wit)
+        .await
+        .is_ok_and(|m| m.is_file()));
+    assert!(tokio::fs::metadata(&test_components_wit)
+        .await
+        .is_ok_and(|m| m.is_file()));
+
+    Ok(())
+}
+
+/// Fail on invalid pull source
+#[tokio::test]
+#[cfg_attr(not(can_reach_ghcr_io), ignore = "ghcr.io is not reachable")]
+async fn test_wash_wit_extended_invalid_source() -> Result<()> {
+    let test_setup = load_fixture("integrated-wkg-extended").await.unwrap();
+
+    let output = test_setup
+        .base_command()
+        .args([
+            "wit",
+            "fetch",
+            "--config-path",
+            "bad-pull-source.wasmcloud.toml",
+        ])
+        .stdout(std::process::Stdio::piped())
+        .stderr(std::process::Stdio::piped())
+        .output()
+        .await
+        .expect("Failed to fetch dependencies");
+    assert!(!output.status.success(), "Failed to fetch dependencies");
+    Ok(())
+}
```

Comment:
Just given some of the history we've had with rolling this out, we may want to have a few other tests like one that uses nested deps (will need to wait for the upstream wkg PR to merge so don't worry about that here) and failure cases for all of the sources:

- not a tarball
- no wit directory
- Non-pinned OCI or git reference
- Pinned OCI or git reference

Reply:
> uses nested deps (will need to wait for the upstream wkg PR to merge so don't worry about that here)

Could you make an issue for this so that we can track it to make sure we get back to it?

I'll work on adding the other tests you noted!

> not a tarball

If we get something that's not a tarball (and trying to unpack it fails), the error message is pretty clear -- are you hoping to catch the specific error we get out in the test?

> no wit directory

Note that the WIT directory is only searched for -- I don't think we need a test for this, because the failure is invalid input (to the upstream functions) at that point (i.e. `wkg` has been pointed to something that just isn't WIT)

Are you hoping to catch a specific error message from the `wkg` side?

> Non-pinned OCI or git reference
> Pinned OCI or git reference

By "pinned" you mean with/out a version specified right?

Right now the tests for the OCI and git references are unpinned -- so was the idea here to add:

- `import ns:pkg/interface` in the WIT
- `ns:pkg@0.2.0` in the config

Is that what you were thinking? Or is this more about enabling version overwrite support?

Reply:
> Could you make an issue for this so that we can track it to make sure we get back to it?

Wkg 0.9 is now out and I think @brooksmtownsend is going to update to it. Maybe the tests there are enough.

> If we get something that's not a tarball (and trying to unpack it fails), the error message is pretty clear -- are you hoping to catch the specific error we get out in the test?

Yeah, just a smoke test to make sure it errors properly

> Right now the tests for the OCI and git references are unpinned 

After out convo above, OCI won't be necessary, but testing a pinned git sha would be good. Maybe a test that pins the right commit and one that pins an older one with incompatible wit?

As for the "no wit directory," I think you're right that it is good enough because the tooling will take care of that. So with that said, I think tests we should add are

- Smoke test for pulling something that isn't a tarball
- Good and bad SHA pins to prove that pinning works

Reply:
Hey added these tests, thanks for the suggestion!